### DISCUSSION TITLE
Split deps from others with a preset

### DISCUSSION THREAD

**Discussion by sclorng** [renovatebot/renovate#13859] [Suggest an Idea]
### What would you like Renovate to be able to do?

Actually, when using semanticPrefixFixDepsChoreOthers, some PR can mix Deps and Others and the type of the commit is non-deterministic.

I would like to be able easily split Deps and Others following the same package rules used to set commit type.

### If you have any ideas on how this should be implemented, please tell us here.

Would it be possible to add a preset like `splitDepsOthers` ? This may be used within a packageRule matching datasource like npm or maven.

```js
splitDepsOthers: {
    description:
      'Splits dependencies from all others',
    packageRules: [
      {
        matchPackagePatterns: ['*'],
        groupName: 'others'
      },
      {
        matchDepTypes: ['dependencies', 'require'],
        groupName: 'deps'
      },
      {
        matchDatasources: ['maven'],
        matchDepTypes: [
          'compile',
          'provided',
          'runtime',
          'system',
          'import',
          'parent',
        ],
        groupName: 'deps'
      },
    ],
  }
```

and used in a package rule like

```json
{
  "matchDatasources": [ "maven", "npm" ],
  "extends": ":splitDepsOthers"
}
```

We may have a version to customize group name

```json
{
  "matchDatasources": [ "maven", "npm" ],
  "extends": ":splitDepsOthers(others, deps)"
}
```

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
I'm not sure I like this idea. What I would prefer is:
- Add tags + matching concept: #4852
- Have each manager populate tags like `isProductionDep` and `isDevDep`
- See if the above can product more elegant `packageRules` structure, instead of embedding so much manager-specific concepts into rules like we're doing today

**Reply by sclorng** [reply]
Yes, I was thinking about something like tags. But I was considering package rule to set them, and being able to match on them in further package rules. Tagging shouldn't be limited to manager only.

Sure, that's definitly a better approach
